nucypher/policy/policies.py

```diff
@@ -217,30 +221,51 @@ def __init__(self,
 
         self._threadpool = ThreadPool(minthreads=threadpool_size, maxthreads=threadpool_size, name=self._repr)
         self.log.info(f"NEM spinning up {self._threadpool}")
+        self._threadpool.callInThread(self._bail_on_timeout)
 
     def __repr__(self):
         return self._repr
 
+    def _bail_on_timeout(self):
+        while True:
+            if self.when_complete.called:
+                return
+            duration = datetime.datetime.now() - self._started
+            if duration.seconds >= self.timeout:
+                try:
+                    self._threadpool.stop()
+                except AlreadyQuit:
+                    raise RuntimeError("Is there a race condition here?  If this line is being hit, it's a bug.")
+                raise RuntimeError(f"Timed out.  Nodes completed: {self.completed}")
+            time.sleep(.5)
+
     def block_until_success_is_reasonably_likely(self):
         """
         https://www.youtube.com/watch?v=OkSLswPSq2o
         """
         if len(self.completed) < self._block_until_this_many_are_complete:
-            completed_for_reasonable_likelihood_of_success = self._partial_queue.get()  # Interesting opportuntiy to pass some data, like the list of contacted nodes above.
+            try:
+                completed_for_reasonable_likelihood_of_success = self._partial_queue.get(timeout=self.timeout) # TODO: Shorter timeout here?
```

Comment:
So we are changing the default here - just want to check that is 20s is a sane default? @jMyles ?

Reply:
I have questions myself about it.  I think it's sane enough for the moment - better to start high (we certainly don't want any policies timing out that weren't actually going to fail).  Let's let the `Alices` come to our channel and complain and then we'll adjust it.